### ISSUE TITLE
Track requirements for Rust nightly

### ISSUE THREAD

**Issue by daviddrysdale** [project-oak/oak#969]
We currently use a fairly recent nightly build of Rust; it would be helpful to track the reasons why we need to do this, e.g.:
 - Support for ThreadSanitizer runs without false positives (#778)

**Comment by daviddrysdale** [comment]
It looks like we only have two things that require nightly channel (see #1036):
 - Running ThreadSanitizer is blocked on https://github.com/rust-lang/rust/issues/39699
 - The `rustfmt` example, which has been disabled since #588 (Feb 2020) anyway (see #594).

So maybe we should delete the `rustfmt` example (I don't think it illustrates anything that isn't already covered by other examples) and do without ThreadSanitizer tests for now.  Then we could use the stable channel.

**Comment by daviddrysdale** [comment]
Current things that require nightly channel (see #1036):
 - Running ThreadSanitizer is blocked on https://github.com/rust-lang/rust/issues/39699.  We could get around this by specifically using nightly Rust for `scripts/run_tests_tsan` (as in #1036).
 - ~~The stable `cargo-doc` doesn't cope with inter-crate links well, and emits some invalid links (blocked on https://github.com/rust-lang/rust/issues/43466).~~ (Appears to no longer be the case with current codebase layout and 1.44.0)
 - Our [`rustfmt` config](https://github.com/project-oak/oak/blob/master/.rustfmt.toml) requires features that are only available in nightly Rust (`comment_width`, `format_code_in_doc_comments`, `imports_layout`, `merge_imports`, `normalize_doc_attributes`, `wrap_comments`).

**Comment by daviddrysdale** [comment]
Also (since #1230):
 - Specifying the output location for built binaries (--out_dir) requires `-Z unstable-options`
